runtime/tchannel_client.go

```diff
@@ -43,150 +47,352 @@ type tchannelClient struct {
 		_ = treader.Close()
-
-		return false, nil, errors.Wrapf(err, "could not read headers for outbound call response: %s", c.serviceName)
+		c.logger.Error("Could not read headers for outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not read headers for outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
 	}
-
 	if err := EnsureEmpty(treader, "reading response headers"); err != nil {
 		_ = treader.Close()
-
-		return false, nil, errors.Wrapf(err, "could not ensure arg2reader is empty for outbound call response: %s", c.serviceName)
+		c.logger.Error("Could not ensure arg2reader is empty for outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not ensure arg2reader is empty for outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
 	}
-
 	if err := treader.Close(); err != nil {
-		return false, nil, errors.Wrapf(err, "could not close arg2reader for outbound call response: %s", c.serviceName)
+		c.logger.Error("Could not close arg2reader for outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not close arg2reader for outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
 	}
 
-	success := !response.ApplicationError()
-	treader, err = response.Arg3Reader()
+	// must have called Arg2Reader before calling ApplicationError
+	c.success = !response.ApplicationError()
+	return nil
+}
+
+// readResHeaders read response headers from arg2
+func (c *tchannelCall) readResBody(response *tchannel.OutboundCallResponse, resp RWTStruct) error {
+	treader, err := response.Arg3Reader()
 	if err != nil {
-		return success, headers, errors.Wrapf(err, "could not create arg3Reader for outbound call response: %s", c.serviceName)
+		c.logger.Error("Could not create arg3Reader for outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not create arg3Reader for outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
 	}
-
 	if err := ReadStruct(treader, resp); err != nil {
 		_ = treader.Close()
+		c.logger.Error("Could not read outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not read outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
+	}
 
-		return success, headers, errors.Wrapf(err, "could not read outbound call response: %s", c.serviceName)
+	structWireValue, err := resp.ToWire()
+	if err != nil {
+		c.logger.Error("Could not serialize response for outbound response", zap.Error(err))
+		return errors.Wrapf(
+			err, "Could not serialize response for outbound %s.%s (%s %s) response",
+			c.client.clientID, c.methodName, c.client.serviceName, c.serviceMethod,
+		)
 	}
+	c.resBody = structWireValue.GetBinary()
```

Comment:
again `GetBinary()` does not do what you think it does. We can just not log the reqBody and resBody for now.

Reply:
Removing `resBody`.